processing/src/main/java/org/apache/druid/query/context/ResponseContext.java

```diff
@@ -155,47 +159,60 @@
     CPU_CONSUMED_NANOS(
         "cpuConsumed",
             (oldValue, newValue) -> (long) oldValue + (long) newValue
+    ),
+    /**
+     * Indicates if a {@link ResponseContext} was truncated during serialization.
+     */
+    TRUNCATED(
+        "truncated",
+            (oldValue, newValue) -> (boolean) oldValue || (boolean) newValue
     );
 
     /**
      * TreeMap is used to have the natural ordering of its keys
      */
-    private static Map<String, BaseKey> map = new TreeMap<>();
+    private static final Map<String, BaseKey> registeredKeys = new TreeMap<>();
 
     static {
       for (BaseKey key : values()) {
-        addKey(key);
+        registerKey(key);
       }
     }
 
     /**
-     * The primary way of registering context keys.
-     * Only the keys registered this way are considered during the context merge.
+     * Primary way of registering context keys.
+     * @throws IllegalArgumentException if the key has already been registered.
      */
-    public static void addKey(BaseKey key)
+    public static void registerKey(BaseKey key)
```

Comment:
Just in case, please make this method `synchronized`

Reply:
added `synchronized`

Reply:
Turns out it was not quite enough, #9106